### PR TITLE
enhance: use package loggers with fields

### PR DESCRIPTION
**Pull request by jbrockopp**
part of the effort for https://github.com/go-vela/community/issues/419

This change updates the server to have a more consistent, standardized set of fields when logging from packages.

This is done by  [using `logrus.WithFields()`](https://pkg.go.dev/github.com/sirupsen/logrus#Entry.WithFields) to output a log message containing relevant fields from the parameters:

https://github.com/go-vela/server/blob/3192c70e36f60ba22ca4b5fe50751d8db18cc3f0/database/postgres/build.go#L23-L28

## Note

Not all functions will have the same level of information available to them so the formatting isn't totally consistent.

The code will make best efforts to log all relevant fields that are available in the content of the function.

Here is a list of some common fields that are logged:

* `build` (number)
* `org` (name)
* `repo` (name)
* `service` (number)
* `step` (number)

## Log Samples

Here is a snippet of what the logs look like, while running a build, before this change:

```
{"level":"debug","msg":"Reading repo JordanBrockopp/vela-play","time":"2021-12-17T15:43:26Z"}
{"level":"trace","msg":"getting repo JordanBrockopp/vela-play from the database","time":"2021-12-17T15:43:26Z"}
{"level":"debug","msg":"Reading build JordanBrockopp/vela-play/1","time":"2021-12-17T15:43:26Z"}
{"level":"trace","msg":"getting build JordanBrockopp/vela-play/1 from the database","time":"2021-12-17T15:43:26Z"}
{"level":"debug","msg":"Verifying user vela-worker has 'write' permissions for repo JordanBrockopp/vela-play","time":"2021-12-17T15:43:26Z"}
{"level":"info","msg":"Updating build JordanBrockopp/vela-play/1","time":"2021-12-17T15:43:26Z"}
{"level":"trace","msg":"updating build 1 in the database","time":"2021-12-17T15:43:26Z"}
{"level":"trace","msg":"getting build JordanBrockopp/vela-play/1 from the database","time":"2021-12-17T15:43:26Z"}
{"api-version":"v0.11.0","body":{"author":"JordanBrockopp","base_ref":"","branch":"main","clone":"https://git.example.com/JordanBrockopp/vela-play.git","commit":"b015d468dcc3606b57e7fc26a7d9ffe3aca441e2","created":1639755805,"deploy":"","deploy_payload":{},"distribution":"linux","email":"jordan_brockopp@example.com","enqueued":1639755806,"error":"","event":"push","finished":0,"head_ref":"","host":"worker","id":1,"link":"http://localhost:8888/JordanBrockopp/vela-play/1","message":"Create .vela.yml","number":1,"parent":1,"ref":"refs/heads/main","repo_id":1,"runtime":"docker","sender":"JordanBrockopp","source":"https://git.example.com/JordanBrockopp/vela-play/commit/b015d468dcc3606b57e7fc26a7d9ffe3aca441e2","started":1639755806,"status":"running","title":"push received from https://git.example.com/JordanBrockopp/vela-play"},"build":1,"fields.time":"2021-12-17T15:43:26Z","ip":"172.20.0.7","latency":8922218,"level":"info","method":"PUT","msg":"","path":"/api/v1/repos/JordanBrockopp/vela-play/builds/1","repo":"JordanBrockopp/vela-play","status":200,"time":"2021-12-17T15:43:26Z","user":"vela-worker","user-agent":"vela-sdk-go/0.11.0"}
```

Here is a snippet of what the logs look like, while running a build, after this change:

```
{"level":"debug","msg":"Reading repo JordanBrockopp/vela-play","time":"2021-12-17T21:45:40Z"}
{"database":"postgres","level":"trace","msg":"getting repo JordanBrockopp/vela-play from the database","org":"JordanBrockopp","repo":"vela-play","time":"2021-12-17T21:45:40Z"}
{"level":"debug","msg":"Reading build JordanBrockopp/vela-play/1","time":"2021-12-17T21:45:40Z"}
{"build":1,"database":"postgres","level":"trace","msg":"getting build JordanBrockopp/vela-play/1 from the database","org":"JordanBrockopp","repo":"vela-play","time":"2021-12-17T21:45:40Z"}
{"level":"debug","msg":"Verifying user vela-worker has 'write' permissions for repo JordanBrockopp/vela-play","time":"2021-12-17T21:45:40Z"}
{"level":"info","msg":"Updating build JordanBrockopp/vela-play/1","time":"2021-12-17T21:45:40Z"}
{"build":1,"database":"postgres","level":"trace","msg":"updating build 1 in the database","time":"2021-12-17T21:45:40Z"}
{"build":1,"database":"postgres","level":"trace","msg":"getting build JordanBrockopp/vela-play/1 from the database","org":"JordanBrockopp","repo":"vela-play","time":"2021-12-17T21:45:40Z"}
{"api-version":"v0.11.0","body":{"author":"JordanBrockopp","base_ref":"","branch":"main","clone":"https://git.example.com/JordanBrockopp/vela-play.git","commit":"b015d468dcc3606b57e7fc26a7d9ffe3aca441e2","created":1639777539,"deploy":"","deploy_payload":{},"distribution":"linux","email":"jordan_brockopp@example.com","enqueued":1639777540,"error":"","event":"push","finished":0,"head_ref":"","host":"worker","id":1,"link":"http://localhost:8888/JordanBrockopp/vela-play/1","message":"Create .vela.yml","number":1,"parent":1,"ref":"refs/heads/main","repo_id":1,"runtime":"docker","sender":"JordanBrockopp","source":"https://git.example.com/JordanBrockopp/vela-play/commit/b015d468dcc3606b57e7fc26a7d9ffe3aca441e2","started":1639777540,"status":"running","title":"push received from https://git.example.com/JordanBrockopp/vela-play"},"build":1,"fields.time":"2021-12-17T21:45:40Z","ip":"172.27.0.5","latency":10887600,"level":"info","method":"PUT","msg":"","path":"/api/v1/repos/JordanBrockopp/vela-play/builds/1","repo":"JordanBrockopp/vela-play","status":200,"time":"2021-12-17T21:45:40Z","user":"vela-worker","user-agent":"vela-sdk-go/0.11.0"}
```

## Full Log File

Here is an example of what the logs look like, while running a build, before this change:

[old.txt](https://github.com/go-vela/server/files/7737942/old.txt)

Here is an example of what the logs look like, while running a build, after this change:

[new.txt](https://github.com/go-vela/server/files/7737953/new.txt)

### CODE REVIEW

File: database/sqlite/secret.go [hunk 1]
```diff
@@ -89,7 +88,7 @@ func (c *client) GetSecret(t, o, n, secretName string) (*library.Secret, error)
 
 // CreateSecret creates a new secret in the database.
 func (c *client) CreateSecret(s *library.Secret) error {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply database/sqlite/secret.go:92]: fixed 👍

File: database/sqlite/secret.go [hunk 2]
```diff
@@ -116,7 +115,7 @@ func (c *client) CreateSecret(s *library.Secret) error {
 
 // UpdateSecret updates a secret in the database.
 func (c *client) UpdateSecret(s *library.Secret) error {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply database/sqlite/secret.go:121]: fixed 👍

File: database/postgres/secret.go [hunk 3]
```diff
@@ -89,7 +88,7 @@ func (c *client) GetSecret(t, o, n, secretName string) (*library.Secret, error)
 
 // CreateSecret creates a new secret in the database.
 func (c *client) CreateSecret(s *library.Secret) error {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply database/postgres/secret.go:92]: fixed 👍

File: database/postgres/secret.go [hunk 4]
```diff
@@ -116,7 +115,7 @@ func (c *client) CreateSecret(s *library.Secret) error {
 
 // UpdateSecret updates a secret in the database.
 func (c *client) UpdateSecret(s *library.Secret) error {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply database/postgres/secret.go:121]: fixed 👍

File: scm/github/webhook.go [hunk 5]
```diff
@@ -75,8 +72,8 @@ func (c *client) VerifyWebhook(request *http.Request, r *library.Repo) error {
 }
 
 // processPushEvent is a helper function to process the push event.
-func processPushEvent(h *library.Hook, payload *github.PushEvent) (*types.Webhook, error) {
-	logrus.Tracef("processing push GitHub webhook for %s", payload.GetRepo().GetFullName())
+func (c *client) processPushEvent(h *library.Hook, payload *github.PushEvent) (*types.Webhook, error) {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply scm/github/webhook.go:77]: fixed 👍

File: scm/github/webhook.go [hunk 6]
```diff
@@ -149,8 +146,8 @@ func processPushEvent(h *library.Hook, payload *github.PushEvent) (*types.Webhoo
 }
 
 // processPREvent is a helper function to process the pull_request event.
-func processPREvent(h *library.Hook, payload *github.PullRequestEvent) (*types.Webhook, error) {
-	logrus.Tracef("processing pull_request GitHub webhook for %s", payload.GetRepo().GetFullName())
+func (c *client) processPREvent(h *library.Hook, payload *github.PullRequestEvent) (*types.Webhook, error) {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply scm/github/webhook.go:153]: fixed 👍

**Review by JordanSussman — APPROVED** [review]

**Comment by KellyMerrick** [comment]
Is there a plan to mitigate all the `This log write receives unsanitized user input from...` linter errors?

**Comment by jbrockopp** [comment]
@KellyMerrick 

AFAICT, those errors are unrelated to my changes and probably out of scope for this 😅 

I believe they are produced from the CodeQL workflow:

https://github.com/go-vela/server/blob/master/.github/workflows/codeql-analysis.yml

And appear to be present in the current codebase, before this change:

https://github.com/go-vela/server/security/code-scanning

If there is a desire to mitigate them, I think a separate story and PR should be created to address them 👍

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Pull request merged**